plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c

```diff
@@ -566,137 +705,104 @@ fu_logitech_bulkcontroller_device_close(FuDevice *device, GError **error)
+	 * Ensure it is running in USB_Device mode
+	 * Response has two data: Request succeeded or failed, and error code in case of failure
+	 */
+	device_request = proto_manager_generate_transition_to_device_mode_request();
 	if (!fu_logitech_bulkcontroller_device_send_sync_cmd(self,
 							     CMD_BUFFER_WRITE,
-							     device_info_request,
+							     device_request,
 							     error)) {
-		g_prefix_error(error, "error in sending buffer write packet: ");
+		g_prefix_error(
+		    error,
+		    "error in sending buffer write packet for transition mode request: ");
 		return FALSE;
 	}
 
-	/* wait for the GetDeviceInfoResponse */
+	/* wait for the TransitionToDeviceModeResponse */
 	if (!fu_logitech_bulkcontroller_device_recv_sync_cmd(self,
 							     CMD_BUFFER_READ,
-							     device_info_response,
+							     device_response,
 							     error)) {
-		g_prefix_error(error, "error in buffer read packet: ");
+		g_prefix_error(error, "error in buffer read packet for transition mode request: ");
 		return FALSE;
 	}
 
 	if (!fu_logitech_bulkcontroller_device_recv_sync_cmd(self,
 							     CMD_UNINIT_BUFFER,
 							     NULL,
 							     error)) {
-		g_prefix_error(error, "error in buffer read packet: ");
+		g_prefix_error(error, "error in buffer read packet for transition mode request: ");
 		return FALSE;
 	}
 	if (!fu_logitech_bulkcontroller_device_send_sync_cmd(self,
 							     CMD_UNINIT_BUFFER,
 							     NULL,
 							     error)) {
-		g_prefix_error(error, "error in sending buffer uninitialize packet: ");
+		g_prefix_error(
+		    error,
+		    "error in sending buffer uninitialize packet for transition mode request: ");
 		return FALSE;
 	}
-	decoded_pkt = proto_manager_decode_message(device_info_response->data,
-						   device_info_response->len,
+	decoded_pkt = proto_manager_decode_message(device_response->data,
+						   device_response->len,
 						   &proto_id,
 						   error);
 	if (decoded_pkt == NULL) {
-		g_prefix_error(error, "error in unpacking packet: ");
+		g_prefix_error(error, "error in unpacking packet for transition mode request: ");
 		return FALSE;
 	}
-	if (proto_id != kProtoId_GetDeviceInfoResponse) {
+	if (proto_id != kProtoId_TransitionToDeviceModeResponse) {
 		g_set_error_literal(error,
 				    G_IO_ERROR,
 				    G_IO_ERROR_INVALID_DATA,
-				    "did not get kProtoId_GetDeviceInfoResponse");
+				    "incorrect response for transition mode request");
 		return FALSE;
 	}
-	if (!fu_logitech_bulkcontroller_device_json_parser(device, decoded_pkt, error))
+	if (!fu_common_read_uint32_safe(decoded_pkt->data,
+					decoded_pkt->len,
+					COMMAND_OFFSET,
+					&success,
+					G_LITTLE_ENDIAN,
+					error))
+		return FALSE;
+	if (!fu_common_read_uint32_safe(decoded_pkt->data,
+					decoded_pkt->len,
+					LENGTH_OFFSET,
+					&error_code,
+					G_LITTLE_ENDIAN,
+					error))
+		return FALSE;
+	g_debug("Received transition mode response. Success: %u, Error: %u", success, error_code);
```

Comment:
maybe hid this behind a `if (g_getenv("FWUPD_LOGITECH_BULKCONTROLLER_VERBOSE") != NULL)`

Reply:
Fixed